File: 351003/Kisel/Rest/Task310RestApi/Task310RestApi/DTOs/Request/NewsRequestTo.cs
using System.ComponentModel.DataAnnotations;

namespace Task310RestApi.DTOs.Request
{
    public class NewsRequestTo
    {
        [Required(ErrorMessage = "Creator ID is required")]
        public long CreatorId { get; set; }

        [Required(ErrorMessage = "Title is required")]
        [StringLength(64, MinimumLength = 2, ErrorMessage = "Title must be between 2 and 64 characters")]
        public required string Title { get; set; }

        [Required(ErrorMessage = "Content is required")]
        [StringLength(2048, MinimumLength = 4, ErrorMessage = "Content must be between 4 and 2048 characters")]
        public required string Content { get; set; }

        public List<long> LabelIds { get; set; } = new List<long>();
    }
}

File: 351004/Shkredova/task310-rest/src/main/java/com/example/task310/dto/NewsRequestTo.java
package com.example.task310.dto;

import jakarta.validation.constraints.Size;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public class NewsRequestTo {

    @NotNull(message = "Creator ID is required")
    private Long creatorId;

    @NotBlank(message = "Title is required")
    @Size(min = 2, max = 64, message = "Title must be between 2 and 64 characters")
    private String title;

    @NotBlank(message = "Content is required")
    @Size(min = 4, max = 2048, message = "Content must be between 4 and 2048 characters")
    private String content;

    public Long getCreatorId() { return creatorId; }
    public void setCreatorId(Long creatorId) { this.creatorId = creatorId; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
}